ext/fileinfo/libmagic/apprentice.c

```diff
@@ -2597,10 +2597,10 @@ getvalue(struct magic_set *ms, struct magic *m, const char **p, int action)
 			convert_libmagic_pattern(&pattern, m->value.s, strlen(m->value.s), options);
 
 			if ((pce = pcre_get_compiled_regex_cache(Z_STR(pattern))) == NULL) {
-				zval_dtor(&pattern);
+				zval_ptr_dtor_nogc(&pattern);
```

Comment:
Same here.

Reply:
Actually, it would be even better if convert_libmagic_pattern were changed to return zend_string*. There's no point in going through a zval here.